akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/BehaviorTestKitImpl.scala

```diff
@@ -172,6 +173,18 @@ private[akka] final class BehaviorTestKitImpl[T](
   override def clearLog(): Unit = context.clearLog()
 
   override def receptionistInbox(): TestInboxImpl[Receptionist.Command] = context.system.receptionistInbox
+
+  override def stopSelf(): Unit = {
+    try {
+      context.setCurrentActorThread()
+      currentUncanonical = Behaviors.stopped
+      current = Behaviors.stopped
+      context.effectQueue.clear()
```

Comment:
I'm not sure the effect queue should be cleared here.  I don't see the reason to prohibit something like:

```scala
// run command and then stop
testkit.run(someCommand)
testkit.stopSelf()

textkit.expectEffectType[Effect.Stopped]
```

I could definitely see how a "Given-When-Then" test might be structured that way ("When I run this command and then stop").

Reply:
The same applies to `clearLog`.